### ISSUE TITLE
Wakeup remote servers on access

### ISSUE THREAD

**Issue by Athos690** [koying/SPMC#597]
Hello and thank you for your beautiful work.
Since version 16.5, on Nvidia Shield TV the function WOL for the servers no longer works (Option Power Saving -> Try to wakeup remote servers on access).
The MAC address search always fails on the server's IP address or name.

**Comment by koying** [comment]
Very possible...

**Comment by koying** [comment]
Please try http://download.semperpax.com/spmc/android-arm/SPMC-16.5.3-spmc-da676e3-armeabi-v7a.apk

**Comment by Athos690** [comment]
Thank-you for your prompt response.
It still does not work, here is an excerpt of the debug.log concerning this operation:

`**16:46:25.393 T:1401755952  NOTICE: WakeOnAccess - Enabled:TRUE
16:46:25.393 T:1534036272   ERROR: OnJobComplete - Mac discovery failed for host '192.168.1.170'
16:46:25.412 T:1401755952   DEBUG: ------ Window Init (DialogKaiToast.xml) ------
16:46:25.422 T:1542392112   ERROR: Unable to lookup host: 'SERVEUR'
16:46:25.423 T:1542392112   ERROR: DoWork - can't determine ip of 'SERVEUR'
16:46:25.423 T:1542392112   ERROR: OnJobComplete - Mac discovery failed for host 'SERVEUR**'

`This works well with a Krypton version and worked well with version 16.4.2 of SPMC with the same server.

If I manually create a file "wakeonlan.xml" the server starts well but it is very long to access the database while the server wakes up in a few seconds only (Windows PC standby S3).

**PS : I do not know if my text is understandable, my language is French and I use Google-Translation to write here ... ;-)**

**Comment by Athos690** [comment]
I rectify :
If I put the "wakeonlan.xml" file created by Kodi Krypton into the "userdata" folder of SPMC version 16.5.3 starts up correctly by launching the server.
It is therefore only the creation of the file "wakeonlan.xml" that does not work.

**Comment by BlindfoldNL** [comment]
My network device (Synology NAS) doesn't go offline, but I was getting a wakeup error anyway when stopping media and going back to the library since installing 1.6.5.1/2.
 I was getting the error that it was trying to wakeup my network device and that either kept waiting, or SPMC crashed after that.
Since I wasn't really having issues with 1.6.4.2 I just reverted to that one until I had time to do some log checks, but when I saw this thread I tried to install this version and I'm haven't gotten the error so far anymore, so it even though it didn't fix Athos's issue, it appears to have fixed something, so thanks and please don't scrap this change in code for future versions ;-)
Will post here  if I notice the error coming back.

**Comment by koying** [comment]
Next try: http://download.semperpax.com/spmc/android-arm/SPMC-16.5.3-spmc-be672c0-armeabi-v7a.apk

**Comment by Athos690** [comment]
It's ok with this latest version !!
The MAC address is found and the "wakeonlan.xml" file is created.

It finds only the MAC address of the IP address and not the one with the name of the server but it is not inconvenient since it is the same and it was the same with the version 16.4.2 and the Krypton version of Kodi..

I think this is due to the PC server because with a Synology NAS he found both addresses.

Anyway, just one address is enough to activate the server ...


Thanks for your responsiveness. :-)

**Comment by BlindfoldNL** [comment]
Also with this latest version I'm not having issues anymore. Thanks for the quick fix koying! (and thanks for reporting it Athos)

**Comment by conorhub** [comment]
I use an old laptop as a video server and was never able to wake it using Kodi. I moved to SPMC some time ago and just saw that the newest update addresses a WOL bug. I tried it and SPMC still reports that the file doesn't exist.

The thing is, all the utilities I've tried to WOL from my other computers or phone cause the laptop to wake as expected. So I'm not sure why this SPMC update didn't change the situation for me.
